builder/azure/arm/template_factory_test.go

```diff
@@ -748,6 +748,41 @@ func TestTrustedLaunch01(t *testing.T) {
 	approvaltests.VerifyJSONStruct(t, deployment.Properties.Template)
 }
 
+func TestSigSourcedWithDiskEncryptionSet(t *testing.T) {
```

Comment:
These two tests panic without the changes to template_builder.go to validate the fix